docs/UserGuide.md

```diff
@@ -1,134 +1,162 @@
+> Example: `add do CS2100 assignment by 10th August`<br>
+> Example: `add go shopping by tomorrow`<br>
+- `add TASK ITEM by [DATE][TIME]` <br>
+This format can be used to add tasks with a stipulated deadline.<br>
+> Example: `add do lab homework on Friday 7pm` <br>
+-  `add TASK ITEM on [DATE]` <br>
+The use of word �on� gives more flexibility for you.<br>
+> Example: `add schedule meeting on Thursday` <br>
+- `add TASK ITEM on [DATE][TIME]` <br>
+> Example: `add meet teacher on Friday 7pm` <br>
+
+> Take Note! Dates need to refer to the current week dates the task was keyed in. Eg. 
+> "tomorrow" and "thursday" is valid but "next Saturday" is invalid. Only the words "on"
+> or "by" has to be used to distinguish between dateline and task description.<br>
+
+#### Listing all tasks : `list`
+To view a list of all the tasks, <kbd>Enter</kbd> list.<br>
+<br>
+Formats: 
+- `list` <br>
+Prints a list of all the uncompleted tasks.<br>
+- `list DATE` <br> 
+Prints a list of all the completed tasks.<br>
+- `list DONE` <br>
+Prints a list of all the tasks due on the specific date.<br>
+<br><img src="images/ListToday.png" width="600"></br>
+Picture 2: List of tasks due today printed when �List today� is keyed in.
+   
+#### Finding tasks: `find`
+To view tasks with similar keyword, use the find command.<br>
+Formats: <br>
+-`find KEYWORD [MORE_KEYWORDS]`<br>
+Prints a list of activities that match the keywords.<br>
+Example: `find banana milk essay:`<br>
+This returns any task description having keywords banana, milk, essay. <br>
+
+-`find-tag KEYWORD `<br>
+Prints list of activities with the same tag.<br>
+ <br><img src="images/findReport.png" width="600"></br>
+ <br><img src="images/findReportResult.png" width="600"></br>
+
+Picture 3 and 4: Keying in �find report� displays list of tasks with report as one of the keywords in task description
+
+> Take Note! <br>
+> * The order of the keywords does not matter. e.g. `chicken egg` will match `egg chicken`.
+> * Full words will be matched e.g. `chicken` will match `chickens`.
+> * Tasks matching at least one keyword will be returned (i.e. `OR` search).
+    e.g. `chicken` will match `chicken duck`
+
+#### Reverting previous action : `undo`
+If you wish to undo your previous action, <kbd>Enter</kbd> undo.<br>
+Format: `undo`
+
+#### Deleting a task : `delete`
+To delete a task, use the delete command. This command deletes the task at a specified INDEX. The index refers to the index number shown in the most recent listing.<br>
+Format: `delete INDEX`<br>
+ <br><img src="images/delete1.png" width="600"></br>
+ <br><img src="images/delete1Result.png" width="600"></br>
+Picture 5 and 6: Entering `delete 1 will delete "Arrange meeting with XYZ company".`<br>
+
+Example: `find violin, then delete 1<br>
+This deletes the 1st task in the results of the find command.
+
+> Take Note! This action can be reversed via undo, only if no new commands are entered 
+> after deleting.
+
+#### Editing a task : `edit`
```

Comment:
your edit command looks half baked. please be very precise with the command format and its behavior.